### DISCUSSION TITLE
Opensearch instead of Elasticsearch

### DISCUSSION THREAD

**Discussion by dcore94** [Netflix/conductor#2936] [Q&A]
Hi all, I've been asked to move from Elasticsearch 7 (which I could setup thanks to your guide) to Opensearch.
Of course the first naif attempt was to change the name of the image in my docker-compose :-). Besides some incompatibilities with parameters which should be no prob, I now get an exception in the server:

`Error creating bean with name 'executionDAOFacade' defined in URL [jar:file:/app/libs/conductor-server-3.8.0-SNAPSHOT-boot.jar!/BOOT-INF/lib/conductor-core-3.8.0-SNAPSHOT.jar!/com/netflix/conductor/core/dal/ExecutionDAOFacade.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es7IndexDAO': Invocation of init method failed; nested exception is org.apache.http.ConnectionClosedException: Connection is closed`

So it looks like the client is not compatible which I supposed to be because the APIs should not change ... . Does anyone have suggestions on how to move ahead from this? And if this is even possible at the moment ... 
Thanks.

**Comment by apanicker-nflx** [comment]
@dcore94 Opensearch supports wire compatiblity with ES7.10.2 whereas the conductor supported version for ES7 is 7.6.2

To use Opensearch at this version you can choose to extend conductor.
This can be made possible by cloning the repo and creating your Opensearch backed implementation of the IndexDAO interface. As can be seen, ES6 and ES7 are self-contained modules which implement this interface. The core module does is not aware of the implementation and uses the API contracts in index dao.

**Reply by markxnelson** [reply]
Hi, is there any update to this? Is there any plan to support OpenSearch?  Or is it still up to users to DIY.
Thanks!!

**Comment by v1r3n** [comment]
Hi @dcore94 we have build that does not use Ealsticsearch for indexing if you want to try that out:
https://github.com/orkes-io/orkes-conductor-community

cc: @markxnelson

**Reply by dcore94** [reply]
Nice to know. Will have a look at it.

**Comment by mrwersa** [comment]
Is there an update for this question? Is there any plan to support AWS opensearch in the future?
